### PR TITLE
feat: new `include_elf!` macro for importing ELF

### PR DESCRIPTION
**Pull request by xJonathanLEI**
Adds a new mechanism of importing ELF files without breaking the existing ELF-copying behaviour. Note that this backward-compatibility is preserved only when building a single target, which is fine as building multiple binary targets doesn't even work at the moment in the first place.

A follow-up PR should add an option to turn off the ELF-copying behaviour.

### CODE REVIEW

**Comment by xJonathanLEI** [comment]
Sure I can make all existing examples use this new syntax.

File: crates/build/src/build.rs [hunk 1]
```diff
@@ -98,3 +112,46 @@ pub(crate) fn build_program_internal(path: &str, args: Option<BuildArgs>) {
...
+    let mut target_elf_paths = vec![];
+
+    for program_crate in metadata.workspace_default_members.iter() {
+        let program = metadata
+            .packages
+            .iter()
+            .find(|p| &p.id == program_crate)
+            .ok_or_else(|| anyhow::anyhow!("cannot find package for {}", program_crate))?;
+
+        for bin_target in program.targets.iter().filter(|t| {
```
[the remark this answers is not in the document]
Comment (xJonathanLEI) [reply crates/build/src/build.rs:131]: Nope. Exactly the opposite. We only care about `bin` targets.

In terms of the `--bin` flag, this is currently arlready supported as an option before this PR. When that flag is used it only builds one target specified by it. So what's done here is just to preserve that behavior.

File: crates/build/src/build.rs [hunk 2]
```diff
@@ -25,7 +25,7 @@ use crate::{
 pub fn execute_build_program(
     args: &BuildArgs,
     program_dir: Option<PathBuf>,
-) -> Result<Utf8PathBuf> {
+) -> Result<Vec<(String, Utf8PathBuf)>> {
```
[the remark this answers is not in the document]
Comment (xJonathanLEI) [reply crates/build/src/build.rs:28]: Oops. Updated to change the paragraph.

File: crates/build/src/build.rs [hunk 3]
```diff
@@ -46,7 +46,15 @@ pub fn execute_build_program(
 
     execute_command(cmd, args.docker)?;
```
[the remark this answers is not in the document]
Comment (xJonathanLEI) [reply crates/build/src/build.rs:47]: Yes it works when building with `cargo prove build --docker`.

**Review by jtguibas — APPROVED** [review]

**Pull request merged**